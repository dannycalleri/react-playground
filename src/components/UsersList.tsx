import React from "react";
import { Link } from "react-router-dom";
import type { User } from "../types";

interface UsersListProps {
  users: User[];
}

export function UsersList(props: UsersListProps) {
  const { users } = props;
  return (
    <ul>
      {users.map((user) => (
        <li key={user.id}>
          <Link to={`/users/${user.id}`}>{user.name}</Link>
        </li>
      ))}
    </ul>
  );
}
