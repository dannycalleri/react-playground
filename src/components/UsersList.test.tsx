import { render } from "@testing-library/react";
import { UsersList } from "./UsersList";

jest.mock("react-router-dom", () => {
  return {
    Link: (props: any) => {
      const { to, children } = props;
      return (
        <>
          <a href={to}>{children}</a>
        </>
      );
    },
  };
});

describe("UsersList", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <UsersList
        users={[
          { id: 1, name: "Daniele" },
          { id: 2, name: "Sandro" },
        ]}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
