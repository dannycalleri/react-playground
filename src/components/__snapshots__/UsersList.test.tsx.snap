// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList should render correctly 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/users/1"
      >
        Daniele
      </a>
    </li>
    <li>
      <a
        href="/users/2"
      >
        Sandro
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
